bindings/python/triplestore/backends/rdflib.py

```diff
@@ -0,0 +1,108 @@
+import warnings
+from typing import TYPE_CHECKING
+
+import rdflib
+from rdflib import BNode, Graph, URIRef
+
+from dlite.triplestore import Literal
+
+if TYPE_CHECKING:  # pragma: no cover
+    from collections.abc import Sequence
+    from typing import Generator
+
+    from dlite.triplestore import Triple
+
+
+def asuri(v):
+    """Help function converting a spo-value to proper rdflib type."""
+    if v is None:
+        return None
+    if isinstance(v, Literal):
+        return rdflib.Literal(v.value, lang=v.lang, datatype=v.datatype)
+    if v.startswith("_:"):
+        return BNode(v)
+    return URIRef(v)
+
+
+def astriple(t):
+    """Help function converting a triple to rdflib triple."""
+    s, p, o = t
+    return asuri(s), asuri(p), asuri(o)
+
+
+class RdflibStrategy:
+    """Triplestore strategy for rdflib."""
```

Comment:
Maybe we should have some examples of usage in the docstring?

Reply:
This class implements the rdflib strategy (backend) and is not intended to be used directly by the users. 

I think it is better to put the usage examples in the module documentation of triplestore.py, which provide the "public" interface exposed to the users. The triplestore module has already some usage examples. Is there something you miss?